pkg/apiserver/registry/stats/antreaclusternetworkpolicystats/rest.go

```diff
@@ -67,7 +68,17 @@ func (r *REST) List(ctx context.Context, options *internalversion.ListOptions) (
 	if !features.DefaultFeatureGate.Enabled(features.AntreaPolicy) {
 		return nil, errors.NewBadRequest("feature AntreaPolicy disabled")
 	}
-	items := r.statsProvider.ListAntreaClusterNetworkPolicyStats()
+	labelSelector := labels.Everything()
+	if options != nil && options.LabelSelector != nil {
+		labelSelector = options.LabelSelector
+	}
+	stats := r.statsProvider.ListAntreaClusterNetworkPolicyStats()
+	items := make([]statsv1alpha1.AntreaClusterNetworkPolicyStats, 0, len(stats))
+	for i := range stats {
+		if labelSelector.Matches(labels.Set(stats[i].Labels)) {
```

Comment:
Could we have labels on stats?

Reply:
We don't set labels on them, you mean copying the labels of corresponding policies?

Reply:
I was thinking whether we skip labelSelector.Matches for resources cannot have labels. But as according to you tests, there is no perf difference, I agree it might not be necessary.

Reply:
Another consideration is labelSelector can use "not equal", so it could select resources that don't have labels even the selector is not empty, always matching the labels can guarantee the correctness.